core/client/src/client.rs

```diff
@@ -1265,7 +1375,9 @@ impl<B, E, Block, RA> backend::AuxStore for Client<B, E, Block, RA>
 		I: IntoIterator<Item=&'a(&'c [u8], &'c [u8])>,
 		D: IntoIterator<Item=&'a &'b [u8]>,
 	>(&self, insert: I, delete: D) -> error::Result<()> {
-		crate::backend::AuxStore::insert_aux(&*self.backend, insert, delete)
+		self.lock_import_and_run(|operation| {
+			self.apply_aux(operation, insert, delete)
```

Comment:
I'm on the fence whether this should lock import.

Reply:
If it doesn't, then if something customized happens outside of block import, it can break consensus.

And note that for consensus we always have the choice to use `apply_aux` and lock import only once, so I think it also shouldn't have impact on performance.

Reply:
could you add a comment to this effect?